NOAA-GSL/ufs-srweather-app
# RRFS_dev1: Fix bug in manage_externals

## DESCRIPTION OF CHANGES: 
For quite some time, manage_externals has not been behaving as many of us had assumed it should. This is a bug fix that should alleviate that discrepancy.

It appears that this issue had already been addressed in the [source repository](https://github.com/ESMCI/manage_externals/blob/master/manic/externals_description.py). Since this code is not tracked, but copied, I believe this fix should be appropriate for our uses. I will also open a PR in the ufs-community/ufs-srweather-app.

## TESTS CONDUCTED: 
Sequence went like this before fix:
- Run `manage_externals/checkout_externals` once on fresh clone.
- Run `manage_externals/checkout_externals -S` to see status of current checkout. Stopped running with `u'CMEPS'` and did nothing. (Same behavior has been observed on prior occasions without the`-S` flag)

Sequence after fix:
- Run `manage_externals/checkout_externals` once on fresh clone.
- Run `manage_externals/checkout_externals -S` to see status of current checkout. Loops through all .gitmodules submodules and then shows clean status of all externals.
- Checkout different branch. Run `manage_externals/checkout_externals` and all submodules are updated to expected hashes for new branches. Note: submodules that may not be tracked in current branch (e.g., python_utilities), will still exist on disk, untracked.
